### DISCUSSION TITLE
Altcover does not detect MsTestV2 tests for a ..Net Framework 4.5 project

### DISCUSSION THREAD

**Discussion by santiagogarcia1** [SteveGilham/altcover#160] [General]
Hi,

I have the following C# projects (.net framework 4.5 - VStudio 2017):

1.  A Class library project (sample class only for Altcover testing purposes) 
2. Test project with MSTestV2 tests for methods defined in point 1.

I already:

- Added via nuPackage, Altcover in project of point 2.
- I have unzipped Altcover nuPackage in my computer in order to use Altcover.exe.

My goal is to get coverage report of my tests.

I run this command:
> AltCover.exe /i=C:\SamplePrjTests_MSTestV2\bin\Debug\ 

The directory "C:\SamplePrjTests_MSTestV2\bin\Debug\" is where project of point 2 generates its dll.

I got the report, however It seems like Altcover did not detect Mstestv2 tests, because there is CERO coverage and I have made tests for all methods in the class of point 1.

Could anyone please help me?

**Comment by SteveGilham** [comment]
The command `AltCover.exe /i=C:\Git\Phoenix\SamplePrjTests_MSTestV2\bin\Debug\` writes the instrumented assemblies to `C:\Git\Phoenix\SamplePrjTests_MSTestV2\bin\Debug\__Instrumented`, so the tests would need to be run on the files in that directory to actually record coverage.

It may be easier to use `AltCover.exe /i=C:\Git\Phoenix\SamplePrjTests_MSTestV2\bin\Debug\ --inplace`, which saves the un-instrumented files to  `C:\Git\Phoenix\SamplePrjTests_MSTestV2\bin\Debug\__Saved`, then modifies the ones in the original directory, where the test runner would most probably be expecting the files it wants.

**Reply by santiagogarcia1** [reply]
Thanks for your quick response, I really appreciate it.

It is fine, additionally I only needed to run the tests in the instrumented folder.

"C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console" "C:\SamplePrjTests_Nutv3\bin\Release\__Instrumented\SamplePrjTests_Nutv3.dll"

**Reply by SteveGilham** [reply]
Happy to help.  To be clear, you are saying that everything is fine now -- is that correct?

**Reply by santiagogarcia1** [reply]
Yes It works as expected. However, for each dll of a test project I have to execute AltCover.... 
Let's say that I need only 1 consolidated coverage.xml for all unit tests related to some dlls of test projects.
How can I do that?
I tried but when I use the same directory to store the coverage.xml, it seems it is overwritten, so at the end I only get on coverage.xml the results of the last dll unit test I executed.
What can I do In order to append all coverage results of some unit tests (related to some dll test projects) in a single coverage.xml?

**Reply by SteveGilham** [reply]
You can give multiple input directory arguments to the one invocation of AltCover, which will instrument all the files to their respective `__Instrumented` directories, taking account of cases where the same assembly is present in multiple locations (instrumenting once and copying); and results in one combined coverage report.  The tests can then be run in sequence, each updating the shared report file.

**Comment by santiagogarcia1** [comment]
Thank you, It works correctly.
I really appreciate your help.
